src/persist-client/src/fetch.rs

```diff
@@ -356,6 +356,23 @@ where
     Ok(part)
 }
 
+/// This represents the lease of a seqno. It's generally paired with some external state,
+/// like a hollow part: holding this lease indicates that we may still want to fetch that part,
+/// and should hold back GC to keep it around.
+///
+/// Generally the state and lease are bundled together, as in [LeasedBatchPart]... but sometimes
```

Comment:
Lazy q: where is this used outside of src/persist-client?

Reply:
Mostly not - though I think one or two of these things are used in `persist-txn`.

Reply:
AFAICT `Lease` is not used outside src/persist-client and could be `pub(crate)`. IIRC the persist-txn stuff uses a method that drops it internally

Reply:
Updated the branch! This required making `into_exchangeable_part``pub(crate)` as well and removing a reference from the docstring.